sway-core/src/semantic_analysis/ast_node/expression/intrinsic_function.rs

```diff
@@ -728,6 +728,134 @@ fn type_check_state_store_or_quad(
+/// Constraints: None.
+///
+/// Signature: `__state_store_quad(key: b256, ptr: raw_ptr, slots: u64) -> bool`
+/// Description: Stores `slots` number of slots (`b256` each) starting at address `ptr` in memory
+///              into storage starting at key `key`.
+/// Constraints: None.
+fn type_check_state_quad(
+    ctx: TypeCheckContext,
+    kind: sway_ast::Intrinsic,
+    arguments: Vec<Expression>,
+    type_arguments: Vec<TypeArgument>,
+    span: Span,
+) -> CompileResult<(ty::TyIntrinsicFunctionKind, TypeId)> {
+    let type_engine = ctx.type_engine;
+    let declaration_engine = ctx.declaration_engine;
+
+    let mut warnings = vec![];
+    let mut errors = vec![];
+
+    if arguments.len() != 3 {
+        errors.push(CompileError::IntrinsicIncorrectNumArgs {
+            name: kind.to_string(),
+            expected: 3,
+            span,
+        });
+        return err(warnings, errors);
+    }
+    if type_arguments.len() > 1 {
+        errors.push(CompileError::IntrinsicIncorrectNumTArgs {
+            name: kind.to_string(),
+            expected: 1,
+            span,
+        });
+        return err(warnings, errors);
+    }
+    let mut ctx = ctx
+        .with_help_text("")
+        .with_type_annotation(type_engine.insert_type(declaration_engine, TypeInfo::Unknown));
+    let key_exp = check!(
+        ty::TyExpression::type_check(ctx.by_ref(), arguments[0].clone()),
+        return err(warnings, errors),
+        warnings,
+        errors
+    );
+    let key_ty = check!(
+        CompileResult::from(
+            type_engine
+                .to_typeinfo(key_exp.return_type, &span)
+                .map_err(CompileError::from)
+        ),
+        TypeInfo::ErrorRecovery,
+        warnings,
+        errors
+    );
+    if !key_ty.eq(&TypeInfo::B256, ctx.engines()) {
+        errors.push(CompileError::IntrinsicUnsupportedArgType {
+            name: kind.to_string(),
+            span,
+            hint: Hint::new("Argument type must be B256, a key into the state storage".to_string()),
+        });
+        return err(warnings, errors);
+    }
+    let mut ctx = ctx
+        .with_help_text("")
+        .with_type_annotation(type_engine.insert_type(declaration_engine, TypeInfo::Unknown));
+    let val_exp = check!(
+        ty::TyExpression::type_check(ctx.by_ref(), arguments[1].clone()),
+        return err(warnings, errors),
+        warnings,
+        errors
+    );
+    let mut ctx = ctx
+        .with_help_text("")
+        .with_type_annotation(type_engine.insert_type(
+            declaration_engine,
+            TypeInfo::UnsignedInteger(IntegerBits::SixtyFour),
+        ));
+    let number_of_slots_exp = check!(
+        ty::TyExpression::type_check(ctx.by_ref(), arguments[2].clone()),
+        return err(warnings, errors),
+        warnings,
+        errors
+    );
+    let type_argument = type_arguments.get(0).map(|targ| {
+        let mut ctx = ctx
+            .with_help_text("")
```

Comment:
again, since we set the help text above on `ctx`, I think this is unnecessary.

Reply:
Good point. I will address this stuff in a follow up PR. I think we do this in a few places in this file.

Reply:
Actually, just addressed this here.